tests/test_empirical_tier2.py

```
@@ -342,22 +342,19 @@ def test_p2_closest_donor_attack(fix: EmpiricalFixture) -> None:
-            f"[P2 CONCERNING] Attack success ratio {ratio:.1f}× > 10× random baseline.",
-            UserWarning,
-            stacklevel=2,
-        )
+    assert attack_rate <= 0.50, (
+        f"[P2 ALARMING] Attack success rate {attack_rate:.3f} > 50% absolute. "
+        "Primary donors are identifiable from synthetic output. "
+        "Privacy threshold exceeded."
+    )
+
+    assert ratio <= 10, (
+        f"[P2 CONCERNING] Attack success ratio {ratio:.1f}× > 10× random baseline. "
```

Ruff (0.15.5)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)